test/cql-pytest/test_materialized_view.py

```diff
@@ -992,3 +992,51 @@ def test_base_clustering_prefix_deletion(cql, test_keyspace):
             # After the deletion, all data should be gone from both base and view
             assert [] == list(cql.execute(f"SELECT c2 FROM {table}"))
             assert [] == list(cql.execute(f"SELECT c2 FROM {mv}"))
+
+# Test deleting an entire base partition, where there is a view with the same or
+# different partition key.  When the view has the same partition key as base,
+# the partition deletion can be done on the base in one update of partition
+# tombstone.  In other cases, a tombstone is generated for each row that
+# corresponds to the deleted base partition.
+def test_base_partition_deletion_with_same_view_partition_key(cql, test_keyspace):
```

Comment:
Did you verify that we exercise the new code path with the test?

Perhaps we could look at some metrics? Do other cql-pytests do that?

Reply:
I verified by adding logs in the code path, and also by doing sstable dump with and without the change
I will try to see if there are relevant metrics

Reply:
I did a pytest removing a partition with 10000 rows and reading the metric scylla_database_total_view_updates_pushed_local before and after
with old code it increases by 100, and with new code by 1
Do you think it should be added as a test?

Reply:
I added a metrics test.